dsls/src/main/scala/ch/epfl/directembedding/test/TestBase.scala

```diff
@@ -9,7 +9,7 @@ case object ValDef extends Exp[Int]
 case object NoArgs extends Exp[Int]
 case class JustTargs[T, U]() extends Exp[(T, U)]
 case class JustArgs(x: Exp[Int]) extends Exp[Int]
-case class ArgsAndTArgs[T, U](t: Exp[T], u: Exp[U]) extends Exp[(T, U)]
+case class ArgsAndTArgs[T, U](t: Exp[_], u: Exp[_]) extends Exp[(T, U)]
```

Comment:
It wasn't necessary, good catch!

I had to use existentials for the nested take and quickly assumed this one needed it too.